scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,390 @@
+     */
+    @Override
+    public void init(Map<String, Serializable> args) throws Exception {
+        if (args.containsKey("ftpHostname")) {
+            ftpHostname = args.get("ftpHostname").toString();
+        }
+        if (args.containsKey("ftpPort")) {
+            try {
+                ftpPort = Integer.parseInt(args.get("ftpPort").toString());
+            } catch (NumberFormatException e) {
+            }
+        }
+        if (args.containsKey("ftpExtractArchive")) {
+            ftpExtractArchive = Boolean.parseBoolean(args.get("ftpExtractArchive").toString());
+        }
+        if (args.containsKey(FTP_LOCAL_FILE_PATH)) {
+            ftpLocalFilePath = args.get(FTP_LOCAL_FILE_PATH) == null ? "" : args.get(FTP_LOCAL_FILE_PATH).toString();
+        }
+        if (args.containsKey("ftpRemoteFilePath")) {
+            ftpRemoteFilePath = args.get("ftpRemoteFilePath").toString();
+        }
+        if (args.containsKey("ftpMode")) {
+            ftpMode = args.get("ftpMode").toString();
+        }
+        ftpUsername = getThirdPartyCredential("FTP_USERNAME");
+        ftpPassword = getThirdPartyCredential("FTP_PASSWORD");
+        if (ftpUsername == null || ftpPassword == null) {
+            throw new IllegalArgumentException("You first need to add your ftp username and password (FTP_USERNAME, FTP_PASSWORD) to the third party credentials");
+        }
+    }
+
+    /**
+     * @see JavaExecutable#execute(TaskResult[])
+     */
+    @Override
+    public Serializable execute(TaskResult... results) throws IOException, IllegalArgumentException {
+
+        FTPClient ftpClient = new FTPClient();
+        // connect and login to the server
+        ftpClient.connect(ftpHostname, ftpPort);
+        int reply = ftpClient.getReplyCode();
+        if (!FTPReply.isPositiveCompletion(reply)) {
+            ftpClient.disconnect();
+            throw new IOException("Exception in connecting to FTP Server");
+        }
+        ftpClient.login(ftpUsername, ftpPassword);
+
+        // use local passive mode to pass firewall
+        ftpClient.enterLocalPassiveMode();
+
+        getOut().println("Connected");
+
+        //FTP mode is GET
+        if (ftpMode.equals("GET")) {
+            FtpGet(ftpClient);
+        }
+        //FTP mode is PUT
+        else if (ftpMode.equals("PUT")) {
+            FtpPut(ftpClient);
+        }
+        // log out and disconnect from the server
+        ftpClient.logout();
+        ftpClient.disconnect();
+
+        getOut().println("Disconnected");
+
+        return 1;
+    }
+
+    private void FtpGet(FTPClient ftpClient) throws IOException, IllegalArgumentException {
+        getOut().println("Importing file(s) from " + ftpRemoteFilePath + " to " + ftpLocalFilePath);
+        FTPFile[] ftpFile = ftpClient.listFiles(ftpRemoteFilePath);
+        if (ftpFile.length == 0)
+            throw new IllegalArgumentException(ftpRemoteFilePath + " not found. Please, enter a valid path.");
```

Comment:
I think (I am not sure) Java Code Convention wants us to always use {} after if condition..

Reply:
done